llumnix/backends/vllm/migrate_backend.py

```diff
@@ -0,0 +1,257 @@
+from abc import ABC, abstractmethod
+from typing import List
+import torch
+
+import ray
+import ray.util.collective as col
+from vllm.logger import init_logger
+from vllm.worker.cache_engine import CacheEngine
+from llumnix.config import MigrationConfig
+
+logger = init_logger(__name__)
+
+class MigrateBackendBase(ABC):
+    @abstractmethod
+    def init_col(self, name, world_size, rank) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def destory_col(self) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def warmup(self) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def migrate_cache(self, src_handle, src_blocks: List[int], dst_blocks: List[int]) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def do_send(self, dst_handle, blocks: List[int]):
+        raise NotImplementedError
+
+    @abstractmethod
+    def do_recv(self, src_handle, blocks: List[int]):
+        raise NotImplementedError
+
+@ray.remote(num_cpus=0)
+class ProxyActor:
+    def exec_method(self, is_driver_worker, handle, *args, **kwargs):
+        try:
+            if is_driver_worker:
+                ret = ray.get(handle.execute_engine_method.remote("execute_worker_method", *args, **kwargs))
+            else:
+                ret = ray.get(handle.execute_method.remote(*args, **kwargs))
+        # pylint: disable=try-except-raise
+        except:
+            raise
+
+        return ret
+
+class RPCMigrateBackend(MigrateBackendBase):
```

Comment:
MigrateBackend -> MigrationBackend, so is migrate_backend.py etc. Except for some functions doing migration, we use 'migration' as part of variable's name.

Reply:
And I think RayRpcMigrationBackend and RayColMigrationBackend might be better, because we also use ray to do rpc.